src/modules/site-v2/base/views/data/data.py

```diff
@@ -57,3 +76,186 @@ def protocols():
+          'publication':          bleach.clean(form.publication.data),
+  
+          # Internally used data
+          'dataset':           'public',
+          'publish_status':    PublishStatus.UPLOADED,
+          'is_bulk_file':      False,
+          'hashed_filename':   hashed_filename,
+        })
+
+        tf.set_user(user)
+
+        # Save the TraitFile object to Datastore
+        tf.save()
+
+        # Seed to Phenotype Metadata SQL table
+        new_trait = PhenotypeMetadata()
+        new_trait.add_trait(tf)
+
+      except Exception as ex:
+        logger.error(f'Failed to create a trait file {form.trait_name_user.data}: {ex}')
+        flash('Failed to submit a form. Please try again later.', 'danger')
+        abort(500)
+      
+      # Save file to GCP bucket
+      species_name = Species.get(form.species.data).name
+      blob_name = f'{MODULE_DB_OPERATIONS_TRAITFILE_PUBLIC_FILEPATH}/{species_name}/{user.name}/{hashed_filename}'
+
+      # Check if the file already exists
+      if check_blob_exists(MODULE_DB_OPERATIONS_BUCKET_NAME, blob_name):
+        flash('File already exists.', 'danger')
+      else:
+        upload_blob_from_file_object(MODULE_DB_OPERATIONS_BUCKET_NAME, form.file.data, blob_name)
+
+      # Reset the file pointer
+      form.file.data.seek(0)
+        
+      try:
+        # Seed the file data to Phenotype Database SQL table
+        with LocalUploadFile(form.file.data, valid_file_extensions=TOOL_INPUT_DATA_VALID_FILE_EXTENSIONS) as file:
+
+          # Validate the file
+          try:
+            validate_file(file, [
+                                  StrainValidator( 'strain', species=tf['species'], force_unique=True, force_unique_msgs={} ),
+                                  NumberValidator( None, accept_float=True, accept_na=True ),
+                                ])
+          except Exception as ex:
+            flash(f'Failed to validate the file: {ex.msg}', 'danger')
+            return render_template('data/submit-trait-form.html', **{
+              # Page Info
+              'title': 'Phenotype Database Trait Submission',
+              'tool_alt_parent_breadcrumb': {"title": "Submit Trait", "url": url_for('data.submit_trait_start')},
+
+              # Data
+              'form': form,
+            })
+          
+          # Parse the trait file
+          trait_data_list = []
+          with open(file) as f:
+            for idx, row in enumerate( csv.reader(f, delimiter='\t') ):
+              if idx == 0:
+                continue
+              else:
+                trait_data = {
+                  'trait_name':  tf['trait_name_user'],
+                  'strain_name': row[0],
+                  'trait_value': row[1],
+                  'metadata_id': tf.name
+                }
+                trait_data_list.append(trait_data)
+
+        trait_data = PhenotypeDatabase()
+        trait_data.add_trait_data(trait_data_list)
+
+      except FileUploadError as ex:
```

Comment:
besides the `FileUploadError` the code above could fail due to other content validation or syntax issues, we might want to add also an `except Exception as e:` to this try block.

Reply:
Pushed in the commit 4ef3096c658aadbd507cd252de95181cc723305d